ui/apps/pmm-compat/src/theme.ts

```diff
@@ -20,19 +18,154 @@ export const changeTheme = async (themeId: 'light' | 'dark'): Promise<void> => {
     newCssLink.rel = 'stylesheet';
     newCssLink.href = config.bootData.assets[newTheme.colors.mode];
     newCssLink.onload = () => {
-      // Remove old css file
-      const bodyLinks = document.getElementsByTagName('link');
-      for (let i = 0; i < bodyLinks.length; i++) {
-        const link = bodyLinks[i];
-
+      // Remove old css file after the new one has loaded to avoid flicker
+      const links = document.getElementsByTagName('link');
+      for (let i = 0; i < links.length; i++) {
+        const link = links[i];
         if (link.href && link.href.includes(`build/grafana.${oldTheme.colors.mode}`)) {
-          // Remove existing link once the new css has loaded to avoid flickering
-          // If we add new css at the same time we remove current one the page will be rendered without css
-          // As the new css file is loading
           link.remove();
         }
       }
     };
     document.head.insertBefore(newCssLink, document.head.firstChild);
   }
 };
+
+/* ---------------------------
+ * Right → left theme wiring
+ * --------------------------*/
+
+// Normalize and apply <html> attributes so CSS-based nav updates immediately
+function applyHtmlTheme(modeRaw: unknown) {
+  const mode: 'light' | 'dark' = String(modeRaw).toLowerCase() === 'dark' ? 'dark' : 'light';
+  const html = document.documentElement;
+  const scheme = mode === 'dark' ? 'percona-dark' : 'percona-light';
+
+  if (html.getAttribute('data-theme') !== mode) {
+    html.setAttribute('data-theme', mode);
+  }
+  if (html.getAttribute('data-md-color-scheme') !== scheme) {
+    html.setAttribute('data-md-color-scheme', scheme);
+  }
+  (html.style as CSSStyleDeclaration).colorScheme = mode;
+
+  return mode;
+}
+
+const isIp = (h: string) => /^\d+\.\d+\.\d+\.\d+$/.test(h);
```

Comment:
The communication between `iframe` and `pmm-ui` has it's place in the `CrossFrameMessenger` (`ui/packages/shared/src/messenger.ts`)